src/autopas/containers/ParticleContainerInterface.h

```diff
@@ -172,10 +172,22 @@ class ParticleContainerInterface {
   virtual void deleteAllParticles() = 0;
 
   /**
-   * Get the total number of particles saved in the container (owned and halo but not dummies).
-   * @return Number of particles in the container.
+   * Get the number of particles with respect to the specified IteratorBehavior.
+   * @warning: Since this function counts the number of the respective particles in the internal particle storage, this
+   * is in O(n) + lock is required. Only use it when it is absolutely necessary to have the exact number of different
+   * particle types like owned or halo. If it is enough to have the whole number of particles (owned + halo + dummy),
+   * the function size() can be used.
+   * @param behavior Behavior of the iterator, see IteratorBehavior.
+   * @return The number of particles with respect to the specified IteratorBehavior.
+   */
+  [[nodiscard]] virtual unsigned long getNumberOfParticles(
```

Comment:
```suggestion
  [[nodiscard]] virtual size_t getNumberOfParticles(
```

Reply:
done in [06385c2](https://github.com/AutoPas/AutoPas/pull/782/commits/06385c264cc416ebc407bb395db2e8d1a70c1cd0)
Should we also do this for `size()` ?

Reply:
technically it doesn't matter but for consistency yes please

Reply:
done in [94c85bf](https://github.com/AutoPas/AutoPas/pull/782/commits/94c85bf7a061dbad59c9a4e2f931bbd52c27b653)